### PR TITLE
WIP SciPy sprint, printing special methods to prompt.

### PR DESCRIPTION
**Pull request by AaronV77**
Added a hook for a special method printing on an object. The method is called xonsh_display.

### CODE REVIEW

File: xonsh/pretty.py [hunk 1]
```diff
@@ -115,6 +115,9 @@ def pretty(obj, verbose=False, max_width=79, newline='\n', max_seq_length=MAX_SE
     """
     Pretty print the object's representation.
     """
+    if hasattr(obj, 'xonsh_display'):
+        obj.xonsh_display()
```
Comment (scopatz) [inline xonsh/pretty.py]: I think this needs to `return obj.xonsh_display()`

**Comment by scopatz** [comment]
Thanks for putting this in @AaronV77!

File: xonsh/pretty.py (hunk 1 above)
Comment (AaronV77) [reply xonsh/pretty.py]: Yes, I copied my example and I had it printing from the xonsh_display().

**Comment by scopatz** [comment]
Thanks @AaronV77!

**Pull request merged**